Sources/Testing/ABI/EntryPoints/EntryPoint.swift

```diff
@@ -259,13 +259,31 @@ public struct __CommandLineArguments_v0: Sendable {
   /// argument, representing the version of the event stream schema to use when
   /// writing events to ``eventStreamOutput``.
   ///
-  /// The corresponding stable schema is used to encode events to the event
-  /// stream. ``ABI/Record`` is used if the value of this property is `0` or
-  /// higher.
+  /// This property is internal because its type is internal. External users of
+  /// this structure can use the ``eventStreamSchemaVersion`` property to get or
+  /// set the value of this property.
+  var eventStreamVersionNumber: ABI.VersionNumber?
+
+  /// The value of the `--event-stream-version` or `--experimental-event-stream-version`
+  /// argument, representing the version of the event stream schema to use when
+  /// writing events to ``eventStreamOutput``.
+  ///
+  /// The value of this property is a 1- or 3-component version string such as
+  /// `"0"` or `"1.2.3"`. The corresponding stable schema is used to encode
+  /// events to the event stream. ``ABI/Record`` is used if the value of this
+  /// property is `"0.0.0"` or higher. The testing library compares components
+  /// individually, so `"1.2"` is less than `"1.20"`.
   ///
   /// If the value of this property is `nil`, the testing library assumes that
   /// the current supported (non-experimental) version should be used.
-  public var eventStreamVersion: Int?
+  public var eventStreamSchemaVersion: String? {
+    get {
+      eventStreamVersionNumber.map { String(describing: $0) }
+    }
+    set {
+      eventStreamVersionNumber = newValue.flatMap { ABI.VersionNumber($0) }
```

Comment:
Could we `preconditionFailure` or something if the passed-in string could not be successfully parsed despite being non-`nil`?

Reply:
Sure.